example-app/app/reducers/index.ts

```diff
@@ -39,14 +40,14 @@ export interface State {
  * These reducer functions are called with each dispatched action
  * and the current or initial state and return a new immutable state.
  */
-export const reducers: ActionReducerMap<State> = {
+export const reducers: ActionReducerMap<State, any> = {
```

Comment:
Not quite happy with `any` but I think this is the only way to do it.
Feedback welcome!

Reply:
Although this will fix router store but users still have to manually add ‘any’ by themself to by pass typescript check in their store. we need a better approach

Reply:
What about the last commit? I think we're getting closer...